legend-pure-m3-core/src/test/java/org/finos/legend/pure/m3/tests/function/base/meta/AbstractTestNewQualifiedProperty.java

```diff
@@ -0,0 +1,54 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.finos.legend.pure.m3.tests.function.base.meta;
+
+import org.apache.commons.lang3.StringUtils;
+import org.eclipse.collections.impl.list.mutable.FastList;
+import org.finos.legend.pure.m3.tests.AbstractPureTestWithCoreCompiled;
+import org.finos.legend.pure.m4.coreinstance.CoreInstance;
+import org.junit.Test;
+
+public class AbstractTestNewQualifiedProperty extends AbstractPureTestWithCoreCompiled
+{
+    @Test
+    public void standardCall()
+    {
+        String[] rawSource = {
+                "function go():Any[*]",
+                "{",
+                    "let classA = 'meta::pure::functions::meta::A'->newClass();",
+                    "let classB = 'meta::pure::functions::meta::B'->newClass();",
+                    "let classC = 'meta::pure::functions::meta::C'->newClass();",
+                    "let qualifiedPropertyFunctionType = ^FunctionType(returnMultiplicity = PureOne, returnType = ^GenericType(rawType = Boolean), parameters = ^VariableExpression(name = 'this', multiplicity = PureOne, genericType = ^GenericType(rawType = Any)));",
+                    "let qualifiedProperty = newQualifiedProperty('a', ^GenericType(rawType=$classB), ^GenericType(rawType=$classA), PureOne, $qualifiedPropertyFunctionType);",
+                    "assert('a' == $qualifiedProperty.name, |'Expected qualified property to have name');",
+                    "assert('a' == $qualifiedProperty.functionName, |'Expected qualified property to have name');",
+                    "assert('B' == $qualifiedProperty.owner.name->toOne(), |'Expected qualified property owner to have name');",
+                    "assert(PureOne == $qualifiedProperty.multiplicity, |'Expected qualified property multiplicity');",
+                    "assert('A' == $qualifiedProperty.genericType.rawType->toOne().name, |'Expected qualified property generic type to have name');",
+                    "assert('QualifiedProperty' == $qualifiedProperty.classifierGenericType.rawType->toOne().name, |'Expected qualified property generic type');",
+                    "let typeArguments = $qualifiedProperty.classifierGenericType.typeArguments;",
+                    "assert(1 == $typeArguments->size(), |'Expected qualified property to have one type argument');",
+                    "assert($typeArguments->toOne().rawType->toOne()->instanceOf(FunctionType), |'Expected qualified property type argument to be instance of FunctionType');",
+                    "assert($qualifiedPropertyFunctionType == $typeArguments->toOne().rawType->toOne(), |'Expected qualified property type argument to be instance of FunctionType');",
+                "}"
+        };
+        String source = StringUtils.join(rawSource, "\n") + "\n";
```

Comment:
I would tend to just write this using standard string concatenation, rather than an array of strings plus StringUtils.join.

Reply:
Yup, thanks have changed. Think I grabbed this from somewhere else...but I prefer the concat too.